File: java-solutions/exercises/week2/Reverse.java
class Reverse {
    public static <T> T[] reverse(T[] arr) {
        int l = 0, r = arr.length - 1;
        while (l < r) {
            T t = arr[l];
            arr[l] = arr[r];
            arr[r] = t;
            l++;
            r--;
        }
        return arr;
    }
}

File: python/exercises/week2/Reverse.py
def reverse(arr: []) -> []:
    l,r = 0, len(arr) - 1
    while l < r:
        arr[l], arr[r] = arr[r], arr[l]
        l += 1
        r -= 1
    return arr

def main():
    ln = input()
    print(" ".join(reverse(ln.split())))

if __name__ == "__main__":
    main()